HeterogeneousCore/AlpakaInterface/interface/devices.h

```diff
@@ -10,30 +9,19 @@
 
 namespace cms::alpakatools {
 
-  namespace detail {
-
-    template <typename TPlatform, typename = std::enable_if_t<alpaka::isPlatform<TPlatform>>>
-    inline std::vector<alpaka::Dev<TPlatform>> enumerate_devices() {
-      using Platform = TPlatform;
-      using Device = alpaka::Dev<Platform>;
-
-      std::vector<Device> devices;
-      uint32_t n = alpaka::getDevCount<Platform>();
-      devices.reserve(n);
-      for (uint32_t i = 0; i < n; ++i) {
-        devices.push_back(alpaka::getDevByIdx<Platform>(i));
-        assert(alpaka::getNativeHandle(devices.back()) == static_cast<int>(i));
```

Comment:
Is this behavior now guaranteed by Alpaka?

Reply:
It's guaranteed for non-SYCL backends.

Reply:
I guess for SYCL backend(s) it is sufficient for now that we know about it, and the few places in our code that currently expect `getNativeHandle()` to return an integer should anyway fail to compile for SYCL?

Reply:
> the few places in our code that currently expect `getNativeHandle()` to return an integer should anyway fail to compile for SYCL?

Yes, since for SYCL `getNativeHandle()` returns an `std::pair<sycl::device, sycl::context>`.